src/mint.rs

```diff
@@ -27,12 +27,17 @@ pub type MintSignatures = BTreeMap<DbcContentHash, (PublicKeySet, NodeSignature)
 
 pub const GENESIS_DBC_INPUT: Hash = Hash([0u8; 32]);
 
+pub trait SpendBook: std::fmt::Debug + Clone + IntoIterator {
+    fn lookup(&self, dbc_hash: &DbcContentHash) -> Option<&DbcTransaction>;
+    fn log(&mut self, dbc_hash: DbcContentHash, transaction: DbcTransaction);
```

Comment:
Lets have these methods return a `Result<_, Self::Error>` and use an associated `Error` type on the trait. As soon as we start doing IO in the spendbook, we'll have to return Results.

(You can use `Error=Infallible` for the `SimpleSpendbook` impl)

Reply:
good idea, will do.

Reply:
ok, I impl'd this following the same pattern as was used for KeyManager.   So there is an Error::SpendBook(String) and I call map_err() to convert from the trait error.